ami/flowchart/library/Scipy.py

```diff
@@ -194,6 +224,105 @@ def addInput(self, **args):
         def to_operation(self, **kwargs):
             return gn.Map(name=self.name()+"_operation", **kwargs, func=FitProc(**self.values))
 
+
+
+    class FitPeakProc():
+
+        def __init__(self, *args, **kwargs):
+            self.model = kwargs['Model']
+            self.use_offset = kwargs['Use offset']
+            self.a_0 = kwargs['Initial amplitude']
+            self.x_0 = kwargs['Initial x0']
+            self.fwhm_0 = kwargs['Initial FWHM']
+            self.c_0 = kwargs['Initial offset']
+            self.func = None
+
+        def get_func(self):
+            if self.model == "Gaussian":
+                sigma_0 = self.fwhm_0 / 2.355
+                if self.use_offset:
+                    p0 = [self.a_0, self.x_0, sigma_0, self.c_0]
+                    func = lambda x, a, mu, sig, c: gaussian_func(x, a, mu, sig) + c
```

Comment:
If you dont use lambdas you can actually do all of this directly in to_operation() itself and dont need FitPeakProc. I dont think it really matters too much though.

Reply:
I think I would need lambdas anyway to add the offset parameter. But that can also be part of the to_operation method anyway...
I was thinking to add an optional x input as you did for the generic fit. Maybe I'll refactor this too.